### PR TITLE
reset pKey when message#key is null

### PR DESCRIPTION
**Pull request by jimjh**
The pKey object is reused across messages, but is not being reset between each read. #getNext should empty pKey when it receives a message without a partition key.

This is kinda tricky to reproduce in a test :( Any ideas?

### CODE REVIEW

File: camus-etl-kafka/src/main/java/com/linkedin/camus/etl/kafka/common/KafkaReader.java [hunk 1]
```diff
@@ -116,6 +116,8 @@ public boolean getNext(EtlKey key, BytesWritable payload, BytesWritable pKey) th
         bytes = new byte[origSize];
         buf.get(bytes, buf.position(), origSize);
         pKey.set(bytes, 0, origSize);
```
Comment (HenryCaiHaiying) [inline camus-etl-kafka/src/main/java/com/linkedin/camus/etl/kafka/common/KafkaReader.java]: Actually when do you see message.key() is null?

Comment (jimjh) [inline camus-etl-kafka/src/main/java/com/linkedin/camus/etl/kafka/common/KafkaReader.java]: It happened to us in production.

In [Message.scala](https://github.com/apache/kafka/blob/0.8.1/core/src/main/scala/kafka/message/Message.scala#L136), one of the constructors sets `key` to `null`.

Comment (HenryCaiHaiying) [inline camus-etl-kafka/src/main/java/com/linkedin/camus/etl/kafka/common/KafkaReader.java]: Can you add a logging when the key is null so we know this situation happened.

And when you set the pKey.setSize(0), does that cause any problems in the remaining code?

Comment (jimjh) [inline camus-etl-kafka/src/main/java/com/linkedin/camus/etl/kafka/common/KafkaReader.java]: Added logging!

Reading the BytesWritable source, it appears that setting size to 0 is the canonical way to reset the data. The length of the actual byte array backing the BytesWritable is its capacity, which does not change.

File: camus-etl-kafka/src/main/java/com/linkedin/camus/etl/kafka/common/KafkaReader.java [hunk 2]
```diff
@@ -116,6 +116,9 @@ public boolean getNext(EtlKey key, BytesWritable payload, BytesWritable pKey) th
...
+      } else {
+        log.debug("Received message with null message.key().");
```
Comment (HenryCaiHaiying) [inline camus-etl-kafka/src/main/java/com/linkedin/camus/etl/kafka/common/KafkaReader.java]: This needs to be a WARN category logging.  And you can also log the 'messageAndOffset' to capture more information.

**Comment by jimjh** [comment]
Done!

**Comment by HenryCaiHaiying** [comment]
Thanks.

**Pull request merged**